lib/class-experimental-wp-widget-blocks-manager.php

```diff
@@ -340,20 +346,34 @@ public static function convert_blocks_to_widget( $blocks ) {
 	 */
 	public static function swap_out_sidebars_blocks_for_block_widgets( $sidebars_widgets_input ) {
 		global $sidebars_widgets;
+		global $wp_customize;
 		if ( null === self::$unfiltered_sidebar_widgets ) {
 			self::$unfiltered_sidebar_widgets = $sidebars_widgets;
 		}
+		$changeset_data = null;
+		if ( is_customize_preview() ) {
```

Comment:
Maybe a function exists check on is_customize_preview provides extra security to not break special sites that may not have this function defined.

Reply:
>It seems we are calling setupWidgetAreas just as a way to replace the blocks with the customizer changeset version. Can't we just do a loop call to updateBlocksInWidgetArea to replace blocks with customizer changeset version, avoiding the need to change this action?

Good catch! This is actually not needed anymore after the changes for supporting share URLs. Changeset data is already rendered on the server.

>Maybe a function exists check on is_customize_preview provides extra security to not break special sites that may not have this function defined.

Yes!

Reply:
>Changeset data is already rendered on the server.

Actually, no, because the store loads from an API that doesn't go through that filter, I'll go with the `updateBlocksInWidgetArea` approach after setup and before the subscription.